src/main/java/com/codeus/winter/config/DefaultBeanFactory.java

```diff
@@ -280,4 +317,31 @@ private Stream<Object> getCollectionDependency(Type parameterType, int valueType
             .filter(bean -> Arrays.stream(bean.getClass().getInterfaces())
                 .anyMatch(interfaceClass -> interfaceClass.isAssignableFrom(dependencyClass)));
     }
+
+    private Object applyPostProcessorsBeforeInitialization(Object bean, String beanName) {
+        Object result = bean;
+        for (BeanPostProcessor postProcessor : postProcessors) {
+            result = applyPostProcessor(postProcessor::postProcessBeforeInitialization, bean, beanName);
+        }
+        return result;
+    }
+
+    private Object applyPostProcessorsAfterInitialization(Object bean, String beanName) {
+        Object result = bean;
+        for (BeanPostProcessor postProcessor : postProcessors) {
+            result = applyPostProcessor(postProcessor::postProcessAfterInitialization, bean, beanName);
+        }
+        return result;
+    }
+
+    private Object applyPostProcessor(BiFunction<Object, String, Object> postProcessorFunction,
+                                      Object bean,
+                                      String beanName) {
+        Object result = postProcessorFunction.apply(bean, beanName);
+        if (result == null) {
+            throw new BeanFactoryException(String.format(
+                    "PostProcessor returned null for bean: %s during post processing", beanName));
+        }
+        return result;
```

Comment:
u can use Optional for such case:
`return Optional.ofNullable(postProcessorFunction.apply(bean, beanName)).orElseThrow(() -> BFE()...)`

Reply:
fixed